src/credentials/GroupDataProvider.h

```diff
@@ -45,22 +45,49 @@ class GroupDataProvider
     struct GroupMapping
     {
         // The endpoint to which a GroupId is mapped.
-        EndpointId endpoint;
+        EndpointId endpoint = 0;
         // The GroupId, which, when received in a message will map to the `endpoint`.
-        GroupId group;
+        GroupId group = 0;
+        // Group name
+        CharSpan name;
+        GroupMapping() = default;
+        GroupMapping(EndpointId eid, GroupId gid, const char * groupName) : endpoint(eid), group(gid)
```

Comment:
This is indeed problematic. Since CharSpan does not own the pointer, it is easier to miss when the pointer get out of scope. I actually found this error in groups-server.cpp. I have made GroupMapping fully owner of the name field, which I think is the most intuitive solution, and prevents many problems.